src/java_bytecode/java_object_factory.cpp

```diff
@@ -1028,6 +1030,24 @@ void java_object_factoryt::gen_nondet_struct_init(
         substruct_in_place);
     }
   }
+
+  // If <class_identifier>.cproverValidate() can be found in the
+  // symbol table, we add a call:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // expr.cproverValidate();
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  const irep_idt validate_method_name =
+    "java::" + as_string(class_identifier) + ".cproverValidate:()V";
+
+  if(symbol_table.has_symbol(validate_method_name))
+  {
+    code_function_callt fun_call;
+    fun_call.function() =
+      symbol_table.lookup(validate_method_name)->symbol_expr();
+    fun_call.arguments().push_back(address_of_exprt(expr));
```

Comment:
Check that the symbol's type matches the expectation (checking it has one parameter with `is_this` set is probably enough)